tensorflow_probability/python/distributions/cauchy_test.py

```diff
@@ -432,6 +432,33 @@ def testAssertsPositiveScaleAfterMutation(self):
       with tf.control_dependencies([scale.assign([1., 2., -3.])]):
         self.evaluate(d.mean())
 
+  def testCauchyCauchyKL(self):
+    batch_size = 6
+    mu_a = np.array([3.0] * batch_size)
+    gamma_a = np.array([1.0, 2.0, 3.0, 1.5, 2.5, 3.5])
+    mu_b = np.array([-3.0] * batch_size)
+    gamma_b = np.array([0.5, 1.0, 1.5, 2.0, 2.5, 3.0])
+
+    c_a = tfd.Cauchy(loc=mu_a, scale=gamma_a, validate_args=True)
+    c_b = tfd.Cauchy(loc=mu_b, scale=gamma_b, validate_args=True)
+
+    kl = tfd.kl_divergence(c_a, c_b)
+    reverse_kl = tfd.kl_divergence(c_b, c_a)
+    kl_val = self.evaluate(kl)
+    reverse_kl_val = self.evaluate(reverse_kl)
+
+    kl_expected = np.log((gamma_a + gamma_b)**2 + (mu_a - mu_b)**2) - np.log(4. * gamma_a * gamma_b)
+
+    x = c_a.sample(int(1e5), seed=test_util.test_seed())
```

Comment:
Let's split these tests up (testing against numpy vs. sampling based test.)

Reply:
I split the test into two functions, one with the "WithAnalytic" and other with the "WithMC" postfix, and moved the numpy and sampling-based tests to them respectively.

Also note: I based the tolerances for the MC tests on the ones for tfd.Normal

Reply:
Oh also, the list of registered KLs is reported here:
https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/kl_divergence

Should I also update this, or is this done automatically / internally after the merge?

Reply:
This will be done automatically.